### PR TITLE
limit cache size by cgroup memory limit

### PR DESCRIPTION
**Pull request by sitano**
Implementation of https://github.com/spacejam/sled/issues/685: a cgroup memory limit is in place and bound cache based on that.

Do we need tests here? (at least if cgroup sys fs is present on the host)

### CODE REVIEW

**Comment by sitano** [comment]
It's ok we print WARN to stdout?

**Comment by spacejam** [comment]
@sitano I would use the `warn!` logging macro there instead of `println!`. Rust also has an `eprintln!` macro for printing error messages to stderr but I think in this case it's fine to just use the built-in logger.

File: crates/pagecache/src/config.rs [hunk 1]
```diff
@@ -688,3 +697,20 @@ impl Config {
...
 }
+
+/// See the Kernel's documentation for more information about this subsystem, found at:
+///  [Documentation/cgroup-v1/memory.txt](https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt)
+#[cfg(target_os="linux")]
+fn get_cgroup_memory_limit() -> u64 {
+    std::fs::File::open("/sys/fs/cgroup/memory/memory.limit_in_bytes")
+        .and_then(read_u64_from)
+        .unwrap_or(0)
+}
```
Comment (spacejam) [inline crates/pagecache/src/config.rs]: In my mind I imagined a more annoying implementation than this, but this looks nice! The one change I would make to this is to return an Option (by replacing `.unwrap_or(0)` with `.ok()` to convert the result to an Option, and then in the caller above instead of checking for if it's above 0, using an `if let Some(mem) = get_cgroup_memory_limit() { ...`

Comment (sitano) [reply crates/pagecache/src/config.rs]: > I imagined a more annoying implementation 

What do you think of? I can abstract it more, reuse `cgroup-rs` or fallback to checking rlimit/as?

Comment (spacejam) [reply crates/pagecache/src/config.rs]: I think the cgroup stuff itself is good!

File: azure-pipelines.yml [hunk 2]
```diff
@@ -71,29 +71,23 @@ steps:
...
+      for crate in $crates; do
+      for target in $targets; do
+        pushd crates/$crate
+        echo "setting up $target..."
+        rustup target add $target
+        echo "checking $target..."
+        cargo check --target $target
+        popd
+      done
+      done
```
Comment (spacejam) [inline azure-pipelines.yml]: could you put this in a separate PR? It causes a conflict with a PR I currently have open which migrates more to using github actions, and I'd like to keep things separate in PR's to make small changes easier to merge.

Comment (sitano) [reply azure-pipelines.yml]: yes, easily
